### PR TITLE
fix: Wrong repo list on Explore page if user already loggin.

### LINKED ISSUE #934: Search bar under "Explore" does not work as expected for repositories when signed in
- Gitea version (or commit ref): 1.0.0+269-g3a91ac51
- Git version: 2.8.3
- Operating system: Linux/official docker images
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist:

On my local instance the search bar under "explore" does not seem to work for repositories when I'm signed in (search for users and organizations does work localy). When trying to reproduce that behavior on try.gitea.io you will notice that the search works if your not sigend in.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/42022369-search-bar-under-explore-does-not-work-as-expected-for-repositories-when-signed-in?utm_campaign=plugin&utm_content=tracker%2F47456670&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F47456670&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by appleboy**
fix #934 and also add unit testing.

User already login.

screen shot 2017-02-22 at 3 19 40 pm

User not login

screen shot 2017-02-22 at 3 19 32 pm

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -1810,41 +1815,51 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, _ in
...
 		if err != nil {
 			return nil, 0, fmt.Errorf("Organization: %v", err)
 		}
 
 		for _, org := range opts.Searcher.Orgs {
-			sess.Or("owner_id = ?", org.ID)
+			ownerIds = append(ownerIds, org.ID)
 		}
+
+		cond = cond.Or(builder.And(builder.Like{"lower_name", opts.Keyword}, builder.In("owner_id", ownerIds)))
```
Comment (lunny) [inline models/repo.go:1846]: Why we also need builder.Like? It seems we have added it before?

Comment (lunny) [reply models/repo.go:1846]: I see.

**Comment by lunny** [comment]
Otherwise LGTM

**Comment by strk** [comment]
LGTM

File: models/repo.go [hunk 2]
```diff
@@ -1809,42 +1814,51 @@ func SearchRepositoryByName(opts *SearchRepoOptions) (repos RepositoryList, _ in
 
 	repos = make([]*Repository, 0, opts.PageSize)
 
+	cond = builder.NewCond()
```
Comment (lunny) [inline models/repo.go]: How about put this line to 1803 as `cond = builder.NewCond()`?

**Comment by lunny** [comment]
let L-G-T-M work

**Pull request merged**